piwik-sdk/src/main/java/org/piwik/sdk/TrackHelper.java

```diff
@@ -0,0 +1,719 @@
+    }
+
+    private TrackHelper(@Nullable TrackMe baseTrackMe) {
+        if (baseTrackMe == null) baseTrackMe = new TrackMe();
+        mBaseTrackMe = baseTrackMe;
+    }
+
+    public static TrackHelper track() {
+        return new TrackHelper();
+    }
+
+    public static TrackHelper track(@NonNull TrackMe base) {
+        return new TrackHelper(base);
+    }
+
+    static abstract class BaseEvent {
+
+        private final TrackHelper mBaseBuilder;
+
+        BaseEvent(TrackHelper baseBuilder) {
+            mBaseBuilder = baseBuilder;
+        }
+
+        TrackMe getBaseTrackMe() {
+            return mBaseBuilder.mBaseTrackMe;
+        }
+
+        @Nullable
+        public abstract TrackMe build();
+
+        public void with(@NonNull PiwikApplication piwikApplication) {
+            with(piwikApplication.getTracker());
+        }
+
+        public void with(@NonNull Tracker tracker) {
+            TrackMe trackMe = build();
+            if (trackMe != null) tracker.track(trackMe);
+        }
+    }
+
+    /**
+     * To track a screenview.
+     *
+     * @param path Example: "/user/settings/billing"
+     * @return an object that allows addition of further details.
+     */
+    public Screen screen(String path) {
+        return new Screen(this, path);
+    }
+
+    /**
+     * Calls {@link #screen(String)} for an activity.
+     * Uses the activity-stack as path and activity title as names.
+     *
+     * @param activity the activity to track
+     */
+    public Screen screen(Activity activity) {
+        String breadcrumbs = ActivityHelper.getBreadcrumbs(activity);
+        return new Screen(this, ActivityHelper.breadcrumbsToPath(breadcrumbs)).title(breadcrumbs);
+    }
+
+    public static class Screen extends BaseEvent {
+        private final String mPath;
+        private final CustomVariables mCustomVariables = new CustomVariables();
+        private String mTitle;
+
+        Screen(TrackHelper baseBuilder, String path) {
+            super(baseBuilder);
+            mPath = path;
+        }
+
+        /**
+         * The title of the action being tracked. It is possible to use slashes / to set one or several categories for this action.
+         *
+         * @param title Example: Help / Feedback will create the Action Feedback in the category Help.
+         * @return this object to allow chaining calls
+         */
+        public Screen title(String title) {
+            mTitle = title;
+            return this;
+        }
+
+        /**
+         * Just like {@link Tracker#setVisitCustomVariable(int, String, String)} but only valid per screen.
+         * Only takes effect when setting prior to tracking the screen view.
+         */
+        public Screen variable(int index, String name, String value) {
+            mCustomVariables.put(index, name, value);
+            return this;
+        }
+
+        @Nullable
+        @Override
+        public TrackMe build() {
+            if (mPath == null) return null;
```

Comment:
I get it, avoiding long call-chains sounds reasonable to me to. If user wants to track some custom request there is `TrackMe` class available.